tensorflow/compiler/jit/xla_cluster_util.cc

```diff
@@ -211,6 +211,32 @@ bool HasResourceInputOrOutput(const Node& node) {
                    DT_RESOURCE) != node.output_types().end();
 }
 
+// Check if a node with a resource input, and the input is from
+// Switch op.
+//
+// This is to help workaround a problem with the following use pattern,
+//     op0 : VarHandleOp
+//     op1 : Switch op0, pred
+//     op2 : ReadVariableOp op0
+//     op3 : ReadVariableOp op1
+//
+//  If op2 and op3 are clustered, the op0 and op1 are inputs,
+//  and they are both considered as resources, and
+//  op1 and op0 happen to point to the same resource, and
+//  this will cause an execution error since the resource is
+//  not allowed to be initialized (locked) twice.
+//
+bool HasResourceInputFromSwitch(const Node& node) {
+  if (std::find(node.input_types().begin(), node.input_types().end(),
+                DT_RESOURCE) != node.input_types().end()) {
+    for (auto edge : node.in_edges()) {
```

Comment:
Probably can be a single pass through the edges?  This is also not precise -- you'd return true if there is one edge that is `DT_RESOURCE` and some other unrelated edge that that is an input from a `Switch`.

Reply:
Yes, the implementation is less precise, 
I will update the implementation to make it more precise.  Since there is no simple interface
to refer an input directly,  the implementation looks more complex.